src/rp2_common/pico_cyw43_driver/cyw43_bus_pio_spi.c

```diff
@@ -330,32 +354,32 @@ int cyw43_spi_transfer(cyw43_int_t *self, const uint8_t *tx, size_t tx_length, u
 
 // Initialise our gpios
 void cyw43_spi_gpio_setup(void) {
-    // Setup WL_REG_ON (23)
-    gpio_init(WL_REG_ON);
-    gpio_set_dir(WL_REG_ON, GPIO_OUT);
-    gpio_pull_up(WL_REG_ON);
+    // Setup CYW43_WL_REG_ON_PIN (23)
+    gpio_init(CYW43_WL_REG_ON_PIN);
+    gpio_set_dir(CYW43_WL_REG_ON_PIN, GPIO_OUT);
+    gpio_pull_up(CYW43_WL_REG_ON_PIN);
 
     // Setup DO, DI and IRQ (24)
-    gpio_init(DATA_OUT_PIN);
-    gpio_set_dir(DATA_OUT_PIN, GPIO_OUT);
-    gpio_put(DATA_OUT_PIN, false);
+    gpio_init(CYW43_DATA_OUT_PIN);
+    gpio_set_dir(CYW43_DATA_OUT_PIN, GPIO_OUT);
+    gpio_put(CYW43_DATA_OUT_PIN, false);
```

Comment:
I guess this implies that the code would need quite a bit of refactoring if `CYW43_DATA_OUT_PIN`, `CYW43_DATA_IN_PIN` and `CYW43_IRQ_PIN` **weren't** all set to the same value? I wonder if it might be worth enforcing that with a compile-time assertion? :thinking:

Reply:
Why would it need "significant" changes? We should allow them to be different. I wouldn't expect it to be too much work?

Reply:
I meant that with the code as-currently-written, only `CYW43_DATA_OUT_PIN` is `gpio_init()`ed, and `gpio_init` isn't (yet) called separately for `CYW43_DATA_IN_PIN` and `CYW43_IRQ_PIN`.
(At least that's how it looks from a quick glance, but I've not studied this code in depth, so perhaps it _does_ already cope with the pins being set to different values?)

Reply:
I have no way of testing different configurations, so I'm closing this as I don't want to make changes which might or might not work.